### PR TITLE
Fixed DAG triggering with parameters

### PR DESCRIPTION
**Pull request by potiuk**
There was a bad cherry-pick when merging #15057 that has not
been caught because of quarantined test.

Fixes: #16810

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by kaxil** [comment]
Looks like we already had a fix: https://github.com/apache/airflow/pull/16648 in main .

File: tests/www/views/test_views_trigger_dag.py
Comment (uranusjr) [inline tests/www/views/test_views_trigger_dag.py:55]: Should we remove this in main as well?

**Review by uranusjr — APPROVED** [review]

File: tests/www/views/test_views_trigger_dag.py
Comment (potiuk) [reply tests/www/views/test_views_trigger_dag.py:55]: Absolutely

Comment (uranusjr) [reply tests/www/views/test_views_trigger_dag.py:55]: OK will send a PR.

Comment (potiuk) [reply tests/www/views/test_views_trigger_dag.py:55]: I can create a PR to main with de-quarantining and we can cherry-pick both PRs instead.

Comment (kaxil) [reply tests/www/views/test_views_trigger_dag.py:55]: This test was flaky for some time, we should probably take a look on how it was doing recently

**Comment by potiuk** [comment]
Ok. Let me merge this one and we can see how the de-quarantining PR is doing.

**Comment by potiuk** [comment]
I think we need to come back to the idea of keeping status of quarantined tests in issues (and some way of nagging the committers to fix them). 

I have high hopes for the new GitHub issues (Private Beta) to be able to it much in much simpler and more manageable way (working with GitHub PM for GitHub issues to enable it for `apache` organisation - hopefully next week). 

The new GitHub issues have better ways to organise them, split into sub-issues and better API for automation of any tasks around those (apparently). https://github.com/features/issues/

File: tests/www/views/test_views_trigger_dag.py
Comment (potiuk) [reply tests/www/views/test_views_trigger_dag.py:55]: Yep. Looking at the status of it . I have disabled the automated status gathering of Flaky tests some time ago as it was not as useful due to timeouts  - but we should likely come back to this.

**Comment by kaxil** [comment]
Meanwhile I have cherry-picked the original commit to both v2-1-stable and v2-1-test

And separately created https://github.com/apache/airflow/pull/16819 to require approvals for mering to v2-1-stable

**Comment by potiuk** [comment]
The test is apparently failing in main even AFTER the #16648 fix in main - so I think it was not flaky - it was marked as quarantined because it just did not work :). Surprisingly it worked for me when I fixed it in v2_1_stable PR. Let's see how this one will do - maybe it is a side-effect of some other tests.

The refactors on test_views do not make it easy to track the origin (I cannot see an issue created for that - I try to always create an issue when marking something as quarantined) but I tracked down when it was marked quarantined - Oct 11 2020

5bc5994c2c1a8f73a644e29e98d3d132c6097ab2

```
-    @pytest.mark.xfail(condition=using_mysql, reason="This test might be flaky on mysql")
+    @pytest.mark.quarantined
     def test_trigger_dag_conf(self):
```

Now. Let's see if I can track how long time ago we expected it to fail :)

**Comment by kaxil** [comment]
Yeah must be a side-effect, as the test passes for me locally

**Comment by potiuk** [comment]
Feb 6 2020: f0c31c597a0d55e4602ca94725704e1cd4a066e3

OK: Seems the test was Never expected to work. It was added as "xfail" from the beginning :)

+    @pytest.mark.xfail(condition=True, reason="This test might be flaky on mysql")
+    def test_trigger_dag_conf(self):

**Comment by potiuk** [comment]
Cosing this as already cherry-picked by @kaxil. Looking at quarantine removal in #16818

**Pull request closed without merging**
